src/thunderbots/software/util/ros_messages.cpp

```diff
@@ -180,5 +178,62 @@ namespace Util
             World world(field, ball, friendly_team, enemy_team);
             return world;
         }
+
+        thunderbots_msgs::World transformWorldMessage(
+            const thunderbots_msgs::World& old_world_msg)
+        {
+            thunderbots_msgs::World new_world_msg;
+
+            new_world_msg.ball = transformBallMessage(old_world_msg.ball);
+            new_world_msg.friendly_team.robots =
+                transformRobotMessages(old_world_msg.friendly_team.robots);
+            new_world_msg.enemy_team.robots =
+                transformRobotMessages(old_world_msg.enemy_team.robots);
+
+            return new_world_msg;
+        }
+
+        thunderbots_msgs::Ball transformBallMessage(
+            const thunderbots_msgs::Ball& old_ball_msg)
+        {
+            thunderbots_msgs::Ball new_ball = old_ball_msg;
+
+            new_ball.position.x = -old_ball_msg.position.x;
+            new_ball.position.y = -old_ball_msg.position.y;
+
+            new_ball.velocity.x = -old_ball_msg.velocity.x;
+            new_ball.velocity.y = -old_ball_msg.velocity.y;
+
+            return new_ball;
+        }
+
+        std::vector<thunderbots_msgs::Robot> transformRobotMessages(
+            const std::vector<thunderbots_msgs::Robot>& old_robot_msgs)
+        {
+            std::vector<thunderbots_msgs::Robot> new_robots;
```

Comment:
make the `new_robots` a copy of the `old_robot_msg` initially, so we only transform and overwrite data we need to. Otherwise the new `new_robots` we return is missing information.